cpp/src/arrow/util/compression_lz4.cc

```diff
@@ -234,14 +242,21 @@ class LZ4Compressor : public Compressor {
   LZ4F_compressionContext_t ctx_ = nullptr;
   LZ4F_preferences_t prefs_;
   bool first_time_;
+
+ private:
+  int compression_level_;
 };
 
 // ----------------------------------------------------------------------
 // Lz4 frame codec implementation
 
 class Lz4FrameCodec : public Codec {
  public:
-  Lz4FrameCodec() : prefs_(DefaultPreferences()) {}
+  Lz4FrameCodec(int compression_level)
+      : compression_level_(compression_level == kUseDefaultCompressionLevel
+                               ? kLZ4DefaultCompressionLevel
+                               : compression_level),
+                               prefs_(DefaultPreferences(compression_level)) {}
```

Comment:
If there's a possibility that `compression_level` could be overridden, it seems that same override should be applied to `prefs_`
```suggestion
                               prefs_(DefaultPreferences(compression_level_)) {}
```

Reply:
Ok changed, thanks